### PR TITLE
framework/st_things: Support pem certificate in st_things

### PR DESCRIPTION
**Pull request by an4967**
- remove duplicated rootCA
- support pem & der certificate in st_things

### CODE REVIEW

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c [hunk 1]
```diff
@@ -226,10 +226,32 @@ static OCStackResult seckey_setup(const char *filename, OicSecKey_t *key, OicEnc
 		return OC_STACK_ERROR;
 	}
 
+	if (encoding == OIC_ENCODING_UNKNOW) {
+		char *file_ext;
+		int filename_len = strlen(filename);
+		file_ext = filename + filename_len;
+
+		while(file_ext >= filename) {
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: check coding rule

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c [hunk 2]
```diff
@@ -226,10 +226,32 @@ static OCStackResult seckey_setup(const char *filename, OicSecKey_t *key, OicEnc
 		return OC_STACK_ERROR;
 	}
 
+	if (encoding == OIC_ENCODING_UNKNOW) {
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: not unknown? Is unknow right?

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c [hunk 3]
```diff
@@ -226,10 +226,32 @@ static OCStackResult seckey_setup(const char *filename, OicSecKey_t *key, OicEnc
...
+
+		while(file_ext >= filename) {
+			if (*file_ext == '.') {
+				file_ext += 1;
+				break;
+			}
+			file_ext--;
+		}
+
+		if (strncmp(file_ext, "PEM", 3) == 0 || strncmp(file_ext, "pem", 3) == 0) {
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: If there is no '.', file_ext pointer will be (filename - 1). At that case, is it right to compare with "PEM"?

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/st_things/things_stack/src/common/framework/things_security_manager.c:234: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/src/common/framework/things_security_manager.c:662: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c (hunk 2 above)
Comment (an4967) [reply framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: @sunghan-chang 'UNKNOW' is right. It is fixed in iotivity release code.

**Comment by an4967** [comment]
@sunghan-chang I applied the coding rule, and corrected the problem.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c (hunk 2 above)
Comment (sunghan-chang) [reply framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: What's the meaning of "fixed"?

Comment (an4967) [reply framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: @sunghan-chang It is using just like that. I can not modify it as the iotivity defined it,

**Pull request merged**